cdk/l3-cdk-resources/atlas-basic/src/index.ts

```diff
@@ -1,98 +1,424 @@
+     * @memberof IpAccessListProps
+     */
+  readonly listOptions?: ipAccessList.ListOptions;
+}
+/**
+ * @description
+ * @export
+ * @interface ApiKeyDefinition
+ */
+export interface ApiKeyDefinition {
+  /**
+     * @description
+     * @type {string}
+     * @memberof ApiKeyDefinition
+     */
+  readonly privateKey?: string;
+  /**
+     * @description
+     * @type {string}
+     * @memberof ApiKeyDefinition
+     */
+  readonly publicKey?: string;
+
+}
+/**
+ * @description
+ * @export
+ * @interface AtlasBasicProps
+ * @extends {cdk.StackProps}
+ */
+export interface AtlasBasicProps extends cdk.StackProps{
+  /**
+     * @description
+     * @type {ApiKeyDefinition}
+     * @memberof AtlasBasicProps
+     */
+  readonly apiKeys : ApiKeyDefinition;
+  /**
+     * @description
+     * @type {ProjectProps}
+     * @memberof AtlasBasicProps
+     */
+  readonly projectProps: ProjectProps;
+  /**
+     * @description
+     * @type {ClusterProps}
+     * @memberof AtlasBasicProps
+     */
+  readonly clusterProps: ClusterProps;
+  /**
+     * @description
+     * @type {DatabaseUserProps}
+     * @memberof AtlasBasicProps
+     */
+  readonly dbUserProps?: DatabaseUserProps;
+  /**
+     * @description
+     * @type {IpAccessListProps}
+     * @memberof AtlasBasicProps
+     */
+  readonly ipAccessListProps?: IpAccessListProps;
+}
+/**
+ * @description
+ * @export
+ * @class AtlasBasic
+ * @extends {cdk.Stack}
+ */
 export class AtlasBasic extends cdk.Stack {
+  /**
+   * Creates an instance of AtlasBasic.
+   * @param {Construct} scope
+   * @param {string} id
+   * @param {AtlasBasicProps} props
+   * @memberof AtlasBasic
+   */
   constructor(scope: Construct, id: string, props: AtlasBasicProps) {
     super(scope, id, props);
-    // Create a new MongoDB Atlas Project
-    // const mProject = new project.CfnProject(this, 'project-'.concat(id), {
-    // //    name: defaults.projectName,
-    //     apiKeys: props.apiKeys,
-    //     ...props.projectProps
-    // });
+    //Create a new MongoDB Atlas Project
+    const mProject = new project.CfnProject(this, 'project-'.concat(id), {
+      apiKeys: props.apiKeys,
+      name: props.projectProps.name || projectDefaults.projectName,
+      ...props.projectProps,
+    });
     // Create a new MongoDB Atlas Cluster and pass project ID
     new atlas.CfnCluster(this, 'cluster-'.concat(id),
       {
-        //    name: defaults.clusterName,
-        //    projectId: mProject.ref,
-        clusterType: defaults.clusterType,
-        replicationSpecs: defaults.replicationSpecs,
-        //    apiKeys: props.apiKeys,
+        apiKeys: props.apiKeys,
+        name: props.clusterProps.name || clusterDefaults.clusterName,
+        projectId: mProject.ref,
+        clusterType: clusterDefaults.clusterType,
         ...props.clusterProps,
       },
     );
     // Create a new MongoDB Atlas Database User
     new user.CfnDatabaseUser(this, 'cfn-db-user-'.concat(id),
```

Comment:
This is the id that shows up in cloudformation so I would omit `cfn` as a prefix.

Reply:
updated.